keras_rs/src/utils/keras_utils.py

```diff
@@ -25,3 +27,16 @@ def clone_initializer(
         return initializer_class.from_config(config)
     # If we get a string or dict, just return as we cannot and should not clone.
     return initializer
+
+
+def check_shape(shape1: types.TensorShape, shape2: types.TensorShape) -> bool:
```

Comment:
Let's give it a slightly more descriptive name, maybe `check_shapes_compatible`? Or `static_dimensions_equal`?

Also add a brief docstring.